test/e2e/operator/nmstate_install_test.go

```diff
@@ -0,0 +1,135 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	v1 "k8s.io/api/apps/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	nmstatev1beta1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1beta1"
+)
+
+func installNMState(nmstate nmstatev1beta1.NMState) {
+	_, err := getNMStateOperator(nmstate)
+	Expect(apierrors.IsNotFound(err)).To(BeTrue(), "NMState CR does not exist yet")
+	err = framework.Global.Client.Create(context.TODO(), &nmstate, &framework.CleanupOptions{})
+	Expect(err).ToNot(HaveOccurred(), "NMState CR created without error")
+}
+
+func uninstallNMState(nmstate nmstatev1beta1.NMState) {
+	_, err := getNMStateOperator(nmstate)
+	Expect(apierrors.IsNotFound(err)).To(BeFalse(), "NMState CR exists")
+	err = framework.Global.Client.Delete(context.TODO(), &nmstate, &client.DeleteOptions{})
+	Expect(err).ToNot(HaveOccurred(), "NMState CR successfully removed")
+}
+
+func getNMStateOperator(nmstate nmstatev1beta1.NMState) (nmstatev1beta1.NMState, error) {
+	err := framework.Global.Client.Get(context.TODO(), types.NamespacedName{Namespace: nmstate.ObjectMeta.Namespace, Name: nmstate.ObjectMeta.Name}, &nmstate)
+	return nmstate, err
+}
+
+func getNMStateHandler(handler v1.DaemonSet) (v1.DaemonSet, error) {
+	err := framework.Global.Client.Get(context.TODO(), types.NamespacedName{Namespace: handler.ObjectMeta.Namespace, Name: handler.ObjectMeta.Name}, &handler)
+	return handler, err
+}
+
+func getNMStateWebhook(webhook v1.Deployment) (v1.Deployment, error) {
+	err := framework.Global.Client.Get(context.TODO(), types.NamespacedName{Namespace: webhook.ObjectMeta.Namespace, Name: webhook.ObjectMeta.Name}, &webhook)
+	return webhook, err
+}
+
+func getNMStateHandlerList(handler v1.DaemonSet) (v1.DaemonSetList, error) {
+	ds := v1.DaemonSetList{}
+	err := framework.Global.Client.List(context.TODO(), &ds, &client.ListOptions{Namespace: handler.ObjectMeta.Namespace})
```

Comment:
Does get DsList work for you? it didn't work for me in nmstate controller unit tests

Reply:
The tests wouldn't pass if it didn't. The unit tests don't have an actual cluster.

Reply:
This prevents from runing this tests at HCO for example since w are going to have multiple DaemonSet at "hyper-converged" you have to filter by label `component=kubernetes-nmtate-handler`